### PR TITLE
Proposal for extra cloaking styles

### PR DESCRIPTION
**Pull request by tom2tom**
Main changes:
add 2 cloaking styles more suitable for longer 'pass-phrase', one of them 'iOS-like'
don't rely on customCloak callback to set new value
 (NB if it were ok to break compatibility, I'd change the callback API, exclude the $element argument)
- various other minor tweaks

### CODE REVIEW

**Comment by ermish** [comment]
Thanks again @tom2tom, for the contributions! 

I noticed this is somewhat behind the current version. Please pull the latest version, merge, then update the pull request. 

The main differences:
- The main file is now `src/jquery-inputcloak.es6.js`.
- Now uses a grunt file for ES6 transpiling and minifying. 
- Readme updates
- Several plugin improvements. 

@tom2tom, for your changes, if you don't want to run the grunt task, just make the changes to the `src/jquery-inputcloak.es6.js` and `Readme`.

**Comment by tom2tom** [comment]
OK, I've ported changes to the repo structure & files that you now have.
As suggested, I've avoided grunting. yuicompressor creates (slightly) smaller min files.
jQuery func .val(value) has no 2nd boolean parameter (so says the API doc, anyway)

File: README.md [hunk 1]
```diff
@@ -1,50 +1,67 @@
...
-
- type: 'all' \('ssn','credit' or 'all'
-   for display respectively like \*\*\*-\*\*-1234,\*1234 or \*\*\*\*,
-   irrelevant if customCloak is used\)<br />
- symbol: '\*' \(any single-char e.g. '\*',\u2022 \(dot\), 'x'\)<br />
- delay: 0 \(mSec delay between blur event and cloak-application\)<br />
- revealOnFocus: true \(or false\)<br />
- cloakOnBlur: true \(or false\)<br />
- customCloak : undefined \(or name of callback, as
+where settings is an object with any of the following members:
```
Comment (ermish) [inline README.md:25]: There's already a set of settings, so we don't need these.

**Comment by ermish** [comment]
- It looks like you still need to do a merge from Ermish -> tom2tom to resolve conflicts. 
- I've added some requested revisions to the file diffs.

File: README.md [hunk 2]
```diff
@@ -1,50 +1,67 @@
...
 ```
-  \)
+### Need more examples?
+
+Head on over [here](http://ermish.github.io/jquery-inputcloak).
 
-Head on over to [github](http://ermish.github.io/jquery-inputcloak) for examples.
+## API
 
+Properties added to each selected element
```
Comment (ermish) [inline README.md:49]: Change to something like

###Available Properties

| Property | Description |
| --- | --- |
| stuff | here |

File: README.md [hunk 3]
```diff
@@ -1,50 +1,67 @@
...
+```
+.settings = object containing default parameters as modified by initiating application
+._defaults = object containing default parameters
+._name = constant name of this plugin
+```
+Methods added to each selected element
```
Comment (ermish) [inline README.md:55]: Change to something like 

###Available Functions

### .cloak()

`js example`

details here.

**Pull request closed without merging**